registry/storage/driver/storagedriver.go

```diff
@@ -92,11 +93,10 @@ type StorageDriver interface {
 	// Delete recursively deletes all objects stored at "path" and its subpaths.
 	Delete(ctx context.Context, path string) error
 
-	// URLFor returns a URL which may be used to retrieve the content stored at
-	// the given path, possibly using the given options.
-	// May return an ErrUnsupportedMethod in certain StorageDriver
-	// implementations.
-	URLFor(ctx context.Context, path string, options map[string]interface{}) (string, error)
+	// RedirectContent returns a URL which the client of the request r may use
+	// to retrieve the content stored at path. Returning the empty string
+	// signals that the request may not be redirected.
+	RedirectContent(r *http.Request, path string) (string, error)
```

Comment:
nit: I think I'd prefer this to be called`ContentURL` so it's more obvious from the name of the func what's returned (yes you can always check godoc)

Reply:
The function's semantics are more (subtle) than just what is returned. It is not simply returning "the" URL for the content. The driver is dictating _whether_ a particular request for content should be redirected, and if so, where to redirect that request to. The s3 and gcs drivers return a different URL depending on the request's HTTP verb. The cloudfront storage middleware decides whether or not to redirect the request by examining the request's source IP.

I had picked the name `RedirectContent` to draw parallels to `GetContent` and `PutContent`, though now looking at it with fresh eyes I realize that the comparison is more likely to confuse than inform readers. What do you think of the name `RedirectURL`?

Reply:
`RedirectURL` works for me 👍